packages/components-js/tests/e2e/puppeteer/specs/heading.e2e.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  getLifecycleStatus,
+  getProperty,
+  selectNode,
+  setContentWithDesignSystem,
+  setProperty,
+  waitForStencilLifecycle,
+} from '../helpers';
+import type { Page } from 'puppeteer';
+import type { HeadingTag, HeadingVariant } from '@porsche-design-system/components/dist/types/bundle';
+
+let page: Page;
+beforeEach(async () => (page = await browser.newPage()));
+afterEach(async () => await page.close());
+
+const initHeading = (opts?: { variant?: HeadingVariant; slot?: string; tag?: HeadingTag }): Promise<void> => {
+  const { variant, slot, tag } = opts;
+
+  const attrs = [
+    variant ? `variant="${typeof variant === 'object' ? JSON.stringify(variant).replace(/"/g, "'") : variant}"` : '',
+    tag ? `tag="${tag}"` : '',
+  ].join(' ');
+
+  return setContentWithDesignSystem(
+    page,
+    `
+    <p-heading ${attrs}>
+      ${slot ? slot : 'Some Heading'}
+    </p-heading>`
+  );
+};
+
+const getHost = () => selectNode(page, 'p-heading');
+
+const getHeadingTagName = (): Promise<string> =>
```

Comment:
Could be refactored to something like `(await getHost()).evaluate(el => el.shadowRoot.querySelector('.root').tagName)`